File: DAM1/Ejercicio_5_T7/src/model/Persona.java
package model;

public abstract class Persona {

    String nombre, apellido;

    public Persona() {
    }

    public Persona(String nombre, String apellido) {
        this.nombre = nombre;
        this.apellido = apellido;
    }


    public void mostrarDatos() {
        System.out.println("Nombre: "+nombre);
        System.out.println("Apellido: "+apellido);
    }


    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }
}

File: DAM2_Pro/PMDM/Tema_1/TrabajoDeHerencia/src/model/Persona.kt
package model

abstract class Persona(var nombre: String, var apellido: String, var dni: String) {
    var telefono : Int? = null
    var correo: String? = null

    constructor(nombre: String, apellido: String, dni: String, telefono: Int, correo: String): this(
        nombre,apellido,dni
    ) {
        this.telefono = telefono
        this.correo = correo
    }

    open fun mostrarDatos() {
        println("El nombre es: $nombre")
        println("El apellido es: $apellido")
        println("El dni es: $dni")
        println("El telefono es: ${telefono ?: "No se ha inizializado"}")
        println("El correo es: ${correo ?: "No se ha inizializado"}")
    }


}